src/manifolds/StiefelCanonicalMetric.jl

```diff
@@ -0,0 +1,153 @@
+@doc raw"""
+    CanonicalMetric <: Metric
+
+The Canonical Metric is name used on several manifolds, for example for the [`Stiefel`](@ref)
+manifold, see for example[^EdelmanAriasSmith1998].
+
+[^EdelmanAriasSmith1998]:
+    > Edelman, A., Ariar, T. A., Smith, S. T.:
+    > _The Geometry of Algorihthms with Orthogonality Constraints_,
+    > SIAM Journal on Matrix Analysis and Applications (20(2), pp. 303–353, 1998.
+    > doi: [10.1137/S0895479895290954](https://doi.org/10.1137/S0895479895290954)
+    > arxiv: [9806030](https://arxiv.org/abs/physics/9806030)
+"""
+struct CanonicalMetric <: RiemannianMetric end
+
+function distance(M::MetricManifold{ℝ,Stiefel{n,k,ℝ},CanonicalMetric}, q, p) where {n,k}
```

Comment:
Since this is the standard definition of `distance`, is there a reason this is limited to the reals?

Reply:
Is it because the metric-related functions below are only defined for the real case? Is there any reason they can't also be generalized to the complex manifold?

Reply:
Exactly, I am just not 100% sure what the complex case would look like. If you have a reference that this all also works for the complex case, we can surely do that.

Reply:
No, I don't have one, just curious.